### ISSUE TITLE
Setting the filter parameters server-side?

### ISSUE THREAD

**Issue by danr777** [Mimetis/Dotmim.Sync#237]
Hello!, is there a way to set the parameters server side (your HTTP method)?  for example want to set the CustomerID based on something looked up by the authorization header on the REST side...

Figured I could modify the WebServerOrchestrator on the controller's POST but it is always Null, is that normal?  screenshot:
<img width="1172" alt="example" src="https://user-images.githubusercontent.com/4326648/76249301-69fb4200-6219-11ea-9bc2-42166729704b.png">

Anyway to do this?

Awesome project by the way!, thank you in advance

**Comment by Mimetis** [comment]
This issue has been fixed with the release of the version `v0.5`.
If you still experiencing the same bug, please comment here, I will re-open the issue.

For the parameter question, yes you can do it. I did not test it, but you can try something like that:

``` cs
[HttpPost]
public async Task Post()
{
    try
    {
        // Get Orchestrator regarding the incoming scope name (from http context)
        var orchestrator = webServerManager.GetOrchestrator(this.HttpContext);

        orchestrator.OnDatabaseChangesSelecting(dcsa =>
        {
            var parameters = dcsa.Context.Parameters;

            if (parameters != null)
            {
                SyncParameter p = parameters["FilterId"];

                if (p != null)
                    p.Value = 12;
            }
        });

        var progress = new SynchronousProgress<ProgressArgs>(pa => Debug.WriteLine($"{pa.Context.SyncStage}\t {pa.Message}"));

        // handle request
        await webServerManager.HandleRequestAsync(this.HttpContext, default, progress);

    }
    catch (Exception ex)
    {
        await WebServerManager.WriteExceptionAsync(this.HttpContext.Response, ex);
    }
}
```

**Comment by danr777** [comment]
Thank you so much!

**Comment by AbderrahmaneAhmam** [comment]
Hello @Mimetis 

> This issue has been fixed with the release of the version `v0.5`. If you still experiencing the same bug, please comment here, I will re-open the issue.
> 
> For the parameter question, yes you can do it. I did not test it, but you can try something like that:
> 
> ```cs
> [HttpPost]
> public async Task Post()
> {
>     try
>     {
>         // Get Orchestrator regarding the incoming scope name (from http context)
>         var orchestrator = webServerManager.GetOrchestrator(this.HttpContext);
> 
>         orchestrator.OnDatabaseChangesSelecting(dcsa =>
>         {
>             var parameters = dcsa.Context.Parameters;
> 
>             if (parameters != null)
>             {
>                 SyncParameter p = parameters["FilterId"];
> 
>                 if (p != null)
>                     p.Value = 12;
>             }
>         });
> 
>         var progress = new SynchronousProgress<ProgressArgs>(pa => Debug.WriteLine($"{pa.Context.SyncStage}\t {pa.Message}"));
> 
>         // handle request
>         await webServerManager.HandleRequestAsync(this.HttpContext, default, progress);
> 
>     }
>     catch (Exception ex)
>     {
>         await WebServerManager.WriteExceptionAsync(this.HttpContext.Response, ex);
>     }
> }
> ```

This solution is not compatible with the current version, especially concerning multi scopes synchronisation.
